### PR TITLE
SB 32

### PR DESCRIPTION
**Pull request by an8ar**
SB-32 and SB-24 tasks. finished property module. did not write all CRUD methods, should be discussed weather they needed or not. Fixed little migration problem

### CODE REVIEW

File: src/modules/admin/admin.service.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+@Injectable()
+export class AdminService {
+  constructor(private readonly prismaService: PrismaService) {}
+
+  async createRole({ name }: CreateRoleDto) {
+    const role = await this.prismaService.role.findUnique({ where: { name } });
+    if (role) {
+      throw new BadRequestException(`Role: ${role.name} already exists`);
+    }
+    return this.prismaService.role.create({ data: { name } });
```
Comment (almaz020702) [inline src/modules/admin/admin.service.ts]: Is there a point to return something in this function?

File: src/modules/admin/dto/create-role.dto.ts
Comment (almaz020702) [inline src/modules/admin/dto/create-role.dto.ts:1]: Add documentation on DTOs

File: src/modules/owner/owner.service.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  public async registration(
+    userData: CreateUserDto,
+    res: Response,
+  ): Promise<AuthResponse> {
+    const user = await this.authService.registration(userData, res);
+    const role = await this.prismaService.role.upsert({
+      where: { name: UserRoleType.PROPERTY_OWNER },
+      update: {},
+      create: { name: UserRoleType.PROPERTY_OWNER },
+    });
```
Comment (almaz020702) [inline src/modules/owner/owner.service.ts:23]: Maybe just use findUnique? I don't think that there is a point  to check whether it exists or not

File: src/modules/property/dto/create-property.dto.ts
Comment (almaz020702) [inline src/modules/property/dto/create-property.dto.ts:1]: Improve documentation pls

Comment (an8ar) [reply src/modules/property/dto/create-property.dto.ts:1]: what to improve there??

File: src/modules/owner/owner.service.ts (hunk 2 above)
Comment (an8ar) [reply src/modules/owner/owner.service.ts:23]: upsert is create if does not exist, if exist just update, if data is the same it is skipping it. High performance, however, could be changed later

File: src/modules/admin/dto/create-role.dto.ts
Comment (an8ar) [reply src/modules/admin/dto/create-role.dto.ts:1]: ok

File: src/modules/admin/admin.service.ts (hunk 1 above)
Comment (an8ar) [reply src/modules/admin/admin.service.ts]: idk, will be changed later, didnt used this module. will change it later

File: src/modules/owner/owner.service.ts (hunk 2 above)
Comment (almaz020702) [reply src/modules/owner/owner.service.ts:23]: Higher performance than findUnique?

File: src/modules/property/dto/create-property.dto.ts
Comment (almaz020702) [reply src/modules/property/dto/create-property.dto.ts:1]: Add examples or smth as in other DTOs

File: src/modules/admin/admin.service.ts (hunk 1 above)
Comment (kabdik) [reply src/modules/admin/admin.service.ts]: It depends, but usually create endpoint does not return anything

File: src/modules/owner/owner.service.ts (hunk 2 above)
Comment (kabdik) [reply src/modules/owner/owner.service.ts:23]: Actually, this upsert will be executed only once, any other calls of the function will not create propertyOwner role. So, it is better to use findUnique. The basic roles may be created in seed I think

File: src/modules/property/dto/create-property.dto.ts
Comment (kabdik) [reply src/modules/property/dto/create-property.dto.ts:1]: I do not think it is very important to add detailed documentation to DTOs, the point is to frontend to understand what is needed

File: src/modules/user/enums/user-role.enum.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export enum UserRoleType {
+  ADMIN = 'admin',
+  PROPERTY_OWNER = 'propertyOwner',
+  CLIENT = 'client',
+  PROPERTYT_STAFF = 'propertyStaff',
+}
```
Comment (kabdik) [inline src/modules/user/enums/user-role.enum.ts:6]: I think it is better to add these roles in seed

**Pull request merged**